Index: src/main/xforms/incubator/address.xml
===================================================================
--- src/main/xforms/incubator/address.xml	(revision 76)
+++ src/main/xforms/incubator/address.xml	Thu Feb 18 16:37:38 CET 2010
@@ -1,97 +1,114 @@
 <?xml version="1.0" encoding="ISO-8859-1"?>
 <div
-      xmlns="http://www.w3.org/1999/xhtml"
-      xmlns:xf="http://www.w3.org/2002/xforms"
-      xmlns:ev="http://www.w3.org/2001/xml-events">
-        <xf:model id="address" ev:event="xforms-revalidate" ev:defaultAction="cancel">
+        xmlns="http://www.w3.org/1999/xhtml"
+        xmlns:xf="http://www.w3.org/2002/xforms"
+        xmlns:ev="http://www.w3.org/2001/xml-events">
+    <xf:model id="address" ev:event="xforms-revalidate" ev:defaultAction="cancel">
 
 
-            <xf:instance id="default">
-                <address xmlns="">
+        <xf:instance id="default">
+            <address xmlns="">
+                <surname/>
+                <name/>
-                    <street/>
-                    <city/>
-                    <zip/>
-                    <country/>
-                </address>
-            </xf:instance>
+                <street/>
+                <city/>
+                <zip/>
+                <country/>
+            </address>
+        </xf:instance>
-            <xf:bind nodeset="zip"  type="integer"/>
 
+        <xf:bind nodeset="surname" readOnly="true()"/>
+        <xf:bind nodeset="name" readOnly="true()"/>
+        <xf:bind nodeset="street" readOnly="true()"/>
+        <xf:bind nodeset="city" readOnly="true()"/>
+        <xf:bind nodeset="zip" readOnly="true()"/>
+        <xf:bind nodeset="country" readOnly="true()"/>
+
+
-            <xf:submission id="s-update-master"
-                           resource="model:master#instance('contact')/data/address[index('addresses')]"
-                           replace="none"
+        <xf:submission id="s-update-master"
+                       resource="model:master#instance('contact')/data/address[index('addresses')]"
+                       replace="none"
-                           method="post"/>
+                       method="post">
+            <xf:message ev:event="xforms-submit-done" level="ephemeral">Subform has updated Master.</xf:message>
+        </xf:submission>
 
 
-        </xf:model>
-        <xf:group appearance="full">
+    </xf:model>
+    <xf:group appearance="full">
-            <xf:label>Subform Address 1</xf:label>
+        <xf:label>Addressdialog Subfrom 1</xf:label>
+        <xf:input ref="surname">
+            <xf:label id="surname">Surname</xf:label>
+        </xf:input>
+        <xf:input ref="name">
+            <xf:label id="name">Name</xf:label>
+        </xf:input>
-            <xf:input ref="street">
-                <xf:label id="street">Street</xf:label>
-            </xf:input>
-            <xf:input ref="city">
+        <xf:input ref="street">
+            <xf:label id="street">Street</xf:label>
+        </xf:input>
+        <xf:input ref="city">
-                <xf:label id="city-lbl">City</xf:label>
+            <xf:label id="city">City</xf:label>
-            </xf:input>
-            <xf:input ref="zip">
+        </xf:input>
+        <xf:input ref="zip">
-                <xf:label id="zip-lbl">Zip</xf:label>
+            <xf:label id="zip">Zip</xf:label>
-                <xf:alert>Please put your ZIP code here</xf:alert>
-            </xf:input>
+            <xf:alert>Please put your ZIP code here</xf:alert>
+        </xf:input>
-            <xf:select1  appearance="minimal" ref="country" id="countries">
+        <xf:select1 appearance="minimal" ref="country" id="countries">
-                <xf:label id="country-lbl">Country</xf:label>
-                <xf:choices id="C-38">
+            <xf:label id="country">Country</xf:label>
+            <xf:choices>
-                    <xf:item selected="false">
-                        <xf:label>Albania</xf:label>
-                        <xf:value>Albania</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>Belgium</xf:label>
-                        <xf:value>Belgium</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>Cuba</xf:label>
-                        <xf:value>Cuba</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>Denmark</xf:label>
-                        <xf:value>Denmark</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>Estonia</xf:label>
-                        <xf:value>Estonia</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>France</xf:label>
-                        <xf:value>France</xf:value>
-                    </xf:item>
-                    <xf:item selected="true">
-                        <xf:label>Germany</xf:label>
-                        <xf:value>Germany</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>Gibraltar</xf:label>
-                        <xf:value>Gibraltar</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>Hungary</xf:label>
-                        <xf:value>Hungary</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>Italy</xf:label>
-                        <xf:value>Italy</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>Japan</xf:label>
-                        <xf:value>Japan</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>Spain</xf:label>
-                        <xf:value>Spain</xf:value>
-                    </xf:item>
-                </xf:choices>
-            </xf:select1>
+                <xf:item selected="false">
+                    <xf:label>Albania</xf:label>
+                    <xf:value>Albania</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>Belgium</xf:label>
+                    <xf:value>Belgium</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>Cuba</xf:label>
+                    <xf:value>Cuba</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>Denmark</xf:label>
+                    <xf:value>Denmark</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>Estonia</xf:label>
+                    <xf:value>Estonia</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>France</xf:label>
+                    <xf:value>France</xf:value>
+                </xf:item>
+                <xf:item selected="true">
+                    <xf:label>Germany</xf:label>
+                    <xf:value>Germany</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>Gibraltar</xf:label>
+                    <xf:value>Gibraltar</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>Hungary</xf:label>
+                    <xf:value>Hungary</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>Italy</xf:label>
+                    <xf:value>Italy</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>Japan</xf:label>
+                    <xf:value>Japan</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>Spain</xf:label>
+                    <xf:value>Spain</xf:value>
+                </xf:item>
+            </xf:choices>
+        </xf:select1>
 
-            <xf:trigger>
-                <xf:label>update master form</xf:label>
-                <xf:send submission="s-update-master"/>
-            </xf:trigger>
-        </xf:group>
+        <xf:trigger>
+            <xf:label>update master form</xf:label>
+            <xf:send submission="s-update-master"/>
+        </xf:trigger>
+    </xf:group>
 </div>
Index: src/main/xforms/incubator/contacts2.xhtml
===================================================================
--- src/main/xforms/incubator/contacts2.xhtml	(revision 76)
+++ src/main/xforms/incubator/contacts2.xhtml	Fri Feb 19 13:49:36 CET 2010
@@ -1,101 +1,191 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <html xmlns="http://www.w3.org/1999/xhtml"
-    xmlns:xf="http://www.w3.org/2002/xforms"
-    xmlns:ev="http://www.w3.org/2001/xml-events"
-        >
-    <head>
-        <title>Embedding a subform with the load action</title>
-        <meta name="author" content="joernt at users.sourceforge.net"/>
-        <meta name="author" content="lars.windauer at chiba-project.org"/>
-        <meta name="description" content="XForms demo for embedding subforms with the load action"/>
+	  xmlns:xf="http://www.w3.org/2002/xforms"
+	  xmlns:ev="http://www.w3.org/2001/xml-events"
+		>
+	<head>
+		<title>Embedding a subform with the load action</title>
+		<meta name="author" content="joernt at users.sourceforge.net"/>
+		<meta name="author" content="lars.windauer at chiba-project.org"/>
+		<meta name="description" content="XForms demo for embedding subforms with the load action"/>
+
+		<link rel="stylesheet" type="text/css" href="../../styles/bf.css"/>
+		<link rel="stylesheet" type="text/css" href="../../resources/styles/reference.css"/>
+		<link rel="stylesheet" type="text/css" href="../../resources/styles/demo.css"/>
+
+		<style type="text/css">
+
+			.contactsHeader {
+				font-weight: bold;
+				font-size: 14pt;
+				margin-bottom: 10px;
+				display: block;
+			}
+
+			.contactsRepeatHeader {
+				margin-bottom: 5px;
+				font-weight: bold;
+			}
+
+			.contactsSubForm {
+				border: 1px solid black;
+				display: block;
+			}
+
+			.xfCompactRepeat {
+				margin-bottom: 15px;
+			}
+
+			.xfRepeatIndex {
+				background-color: lightsteelblue;
+			}
+
+			.xfRepeatIndex .xfValue {
+				background-color: lightsteelblue;
+			}
+
+			#xforms {
+				margin-left: auto;
+				margin-right: auto;
+				width: 720px;
+			}
+
+			#addressGroup {
+				padding-top: 15px;
+				display: block;
+			}
+
+			.tundra * {
+				font-size: 12px;
+			}
+		</style>
+
+		<script type="text/javascript">
+			dojo.require('dijit.layout.ContentPane');
+		</script>
-    </head>
+	</head>
-    <body>
-    <form action="">
+<body class="soria" style="margin:30px;">
+		<div id="xforms">
+			<div style="display:none">
-        <xf:model id="master">
+				<xf:model id="master">
-            <xf:instance id="contact" xmlns="">
-                <data>
+					<xf:instance id="contact">
+						<data xmlns="">
-                    <address id="1">
+							<address id="1">
+								<title>Herr</title>
+								<firstNames>Max</firstNames>
+								<name>Mustermann</name>
-                        <street>Goethestr. 47</street>
-                        <city>Hamburg</city>
-                        <zip>223456</zip>
-                        <country>Germany</country>
-                    </address>
-                    <address id="2">
+								<street>Goethestr. 47</street>
+								<city>Hamburg</city>
+								<zip>223456</zip>
+								<country>Germany</country>
+							</address>
+							<address id="2">
+								<title>Frau</title>
+								<firstNames>Maxi</firstNames>
+								<name>Musterfrau</name>
-                        <street>Schillerstr. 42</street>
-                        <city>Berlin</city>
-                        <zip>12234</zip>
-                        <country>Germany</country>
-                    </address>
-                    <address id="3">
+								<street>Schillerstr. 42</street>
+								<city>Berlin</city>
+								<zip>12234</zip>
+								<country>Germany</country>
+							</address>
+							<address id="3">
+								<title>Prof. Dr.</title>
+								<firstNames>MŠxle</firstNames>
+								<name>Mustermann</name>
-                        <street>Passeig de Grˆcia 12</street>
-                        <city>Barcelona</city>
-                        <zip>08008</zip>
-                        <country>Spain</country>
-                    </address>
-                    <address id="4">
+								<street>Passeig de Grˆcia 12</street>
+								<city>Barcelona</city>
+								<zip>08008</zip>
+								<country>Spain</country>
+							</address>
+							<address id="4">
+								<title>Dipl. Inf</title>
+								<firstNames>Max</firstNames>
+								<name>Power</name>
-                        <street>Hegelweg. 23</street>
-                        <city>Dresden</city>
-                        <zip>01326</zip>
-                        <country>Germany</country>
-                    </address>
-                </data>
-            </xf:instance>
+								<street>Hegelweg. 23</street>
+								<city>Dresden</city>
+								<zip>01326</zip>
+								<country>Germany</country>
+							</address>
+						</data>
+					</xf:instance>
 
-            <xf:submission id="update-subform"
-                           resource="model:address#instance('default')/address"
-                           method="post"
-                           replace="none"
-                           ref="address[index('addresses')]">
+					<xf:submission id="update-subform"
+								   resource="model:address#instance('default')/address"
+								   method="post"
+								   replace="none"
+								   ref="address[index('addresses')]">
+						<xf:message ev:event="xforms-submit-done" level="ephemeral">Masterform has updated Subform.
+						</xf:message>
-            </xf:submission>
+					</xf:submission>
 
-        </xf:model>
+				</xf:model>
+			</div>
+			<div class="Section" dojotype="dijit.layout.ContentPane">
 
-        <xf:group appearance="full" id="contact">
-            <xf:label>Address Collection:</xf:label>
+				<xf:group appearance="full" id="contact" class="contactsFullGroup">
+					<div class="Headline">Address Collection:</div>
 
+					<div class="description">
+						This Demo shows the loading of embeded Forms and how the data for the Subforms could be gathered.<br/>
+						The "Down"-Button loads a Subform which data is updated by the Masterform.<br/>
+						The "Up"-Button loads a Subform which updates its data for itself.
+					</div>
 
-			<xf:repeat id="addresses" nodeset="address" appearance="compact">
+					<xf:repeat id="addresses" nodeset="address" appearance="compact" class="contactsRepeat">
+						<xf:output ref="title">
+							<xf:label class="contactsRepeatHeader">Title</xf:label>
+						</xf:output>
+						<xf:output ref="firstNames">
+							<xf:label class="contactsRepeatHeader">First names</xf:label>
+						</xf:output>
+						<xf:output ref="name">
+							<xf:label class="contactsRepeatHeader">Name</xf:label>
+						</xf:output>
-				<xf:output ref="street">
+						<xf:output ref="street">
-					<xf:label>Street</xf:label>
+							<xf:label class="contactsRepeatHeader">Street</xf:label>
-				</xf:output>
-				<xf:output ref="zip">
+						</xf:output>
+						<xf:output ref="zip">
-					<xf:label>ZIP Code</xf:label>
+							<xf:label class="contactsRepeatHeader">ZIP Code</xf:label>
-				</xf:output>
-				<xf:output ref="city">
+						</xf:output>
+						<xf:output ref="city">
-					<xf:label>City</xf:label>
+							<xf:label class="contactsRepeatHeader">City</xf:label>
-				</xf:output>
-				<xf:output ref="country">
+						</xf:output>
+						<xf:output ref="country">
-					<xf:label>Country</xf:label>
+							<xf:label class="contactsRepeatHeader">Country</xf:label>
-				</xf:output>
+						</xf:output>
+
-			</xf:repeat>
+					</xf:repeat>
+				</xf:group>
 
-            <xf:trigger>
-                <xf:label>load selected address [master updates subform]</xf:label>
+				<xf:group appearance="minimal" class="contactsTriggerGroup">
+					<xf:trigger class="contactsMasterTrigger">
+						<xf:label>load selected address</xf:label>
-                <xf:action>
-                    <xf:load show="embed" targetid="address">
-                        <xf:resource value="'address.xml'"/>
-                    </xf:load>
-                    <xf:send submission="update-subform"/>
-                </xf:action>
+						<xf:action>
+							<xf:load show="embed" targetid="address">
+								<xf:resource value="'address.xml'"/>
+							</xf:load>
+							<xf:send submission="update-subform"/>
+						</xf:action>
-
-
-            </xf:trigger>
+					</xf:trigger>
 
-
-            <xf:trigger>
-                <xf:label>load selected address [subforms updates itself]</xf:label>
+					<xf:trigger class="contactsSubTrigger">
+						<xf:label>load selected address</xf:label>
-                <xf:action>
-                    <xf:load show="embed" targetid="address">
-                        <xf:resource value="'address2.xml'"/>
-                    </xf:load>
-                </xf:action>
-            </xf:trigger>
+						<xf:action>
+							<xf:load show="embed" targetid="address">
+								<xf:resource value="'address2.xml'"/>
+							</xf:load>
+						</xf:action>
+					</xf:trigger>
+				</xf:group>
-            
+
-            <div style="border:1px solid black;display:block;">
+
+				<xf:group appearance="full" class="contactsFullGroup">
+					<div class="contactsSubForm">
-				<div id="address"></div>
-			</div>
-        </xf:group>
+						<div id="address"></div>
+					</div>
+				</xf:group>
-    </form>
 
+			</div>
+		</div>
-    </body>
+	</body>
 </html>
Index: src/main/xforms/demo/Atom.xhtml
===================================================================
--- src/main/xforms/demo/Atom.xhtml	(revision 9)
+++ src/main/xforms/demo/Atom.xhtml	Thu Feb 18 21:42:00 CET 2010
@@ -1,157 +1,152 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!--
-    author: Fabian Otto <fabian.otto<AT>gmail.com> 
-    date:   2009-03-03
+    author: Fabian Otto <fabian.otto<AT>betterform.de>
+    author: Tobi Krebs <tobias.krebs<AT>betterform.de>
+    date:   2010-02-18
 -->
 
 <html xmlns="http://www.w3.org/1999/xhtml"
-      xmlns:xf="http://www.w3.org/2002/xforms"
-      xmlns:ev="http://www.w3.org/2001/xml-events"
-      xmlns:xs="http://www.w3.org/2001/XMLSchema"
-      xmlns:atom="http://www.w3.org/2005/Atom"
-      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+	  xmlns:xf="http://www.w3.org/2002/xforms"
+	  xmlns:ev="http://www.w3.org/2001/xml-events"
+	  xmlns:xs="http://www.w3.org/2001/XMLSchema"
+	  xmlns:atom="http://www.w3.org/2005/Atom"
+	  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
 <head>
-    <title>Feed Atom Reader</title>
+	<title>Feed Atom Reader</title>
-    <style type="text/css">
-        #meta-data{
-            display:block;
-            background:lightsteelblue;
-            border:thin outset steelblue;
-            padding:20px;
-        }
-        #meta-data-label{
-            font-size:12pt;
-        }
-        #meta-data .xfLabel{font-size:12pt;}
-        .subtitle .xfValue {
-            font-weight:bolder;
-            padding:5px;
-            font-size:12pt;
-        }
-        .heading {
-            font-weight:bold;
-            padding:10px 0px;
-            font-size:large;
-            display:block;
-        }
-        .date .xfValue {
-             padding:5px;
-        }
-        .time .xfValue {
-             padding:5px;
-        }
-        .xfMinimalTrigger .xfValue {
-            padding-top:5px;
-        }
-        .xfFullGroup .xfRepeatItem{
-            padding:20px;
-            border:thin dotted steelblue;
-        }
-        .xfRepeatItem .subtitle .xfValue{
-            font-size:12pt;
-        }
-        .articleLink{ padding:5px;}
-        .summary .xfValue{display:block;margin:10px 5px;}
-        .xfRepeatItem{background:#dedeff;}
-        .xfRepeatItem .xfValue{background:#dedeff;}
-        .xfRepeatIndexMouseOver{background:#dedeff;}
-        #reload-value{background:url(../../resources/images/reload.gif) no-repeat center; height:30px;width:30px;}
-        .tundra .dijitButtonNode{
-            background-image:none;
-            position:absolute;
-            right:30px;
-            top:44px;
-            
+
-        }
-        .tundra .dijitButtonHover{background-image:none;}
-    </style>
+	<link rel="stylesheet" type="text/css" href="../../styles/bf.css"/>
+	<link rel="stylesheet" type="text/css" href="../../resources/styles/reference.css" />
+	<link rel="stylesheet" type="text/css" href="../../resources/styles/demo.css"/>
+</head>
 
+<body class="soria" style="margin:30px;">
+	<div id="xforms">
+		<div style="display:none">
-    <xf:model id="m-atom">
+			<xf:model id="m-atom">
-        <xf:instance id="i-controller" xmlns="">
-            <data/>
+				<xf:instance id="i-controller">
+					<data xmlns="">
+						<feedURL>http://rss.golem.de/rss.php?feed=ATOM1.0</feedURL>
+					</data>
-        </xf:instance>
+				</xf:instance>
 
-        <xf:instance id="i-atom-entries" xmlns="">
-            <feed/>
+				<xf:instance id="i-atom-entries">
+					<feed xmlns=""/>
-        </xf:instance>
+				</xf:instance>
 
-        <xf:submission id="s-get-data"
-                       ref="instance('i-controller')"
+				<xf:submission id="s-get-data"
+							   ref="instance('i-controller')"
-                       resource="http://rss.golem.de/rss.php?feed=ATOM1.0"
-                       replace="instance"
-                       validate="false"
-                       method="get"
-                       mediatype="application/atom+xml"
-                       instance="i-atom-entries">
+							   replace="instance"
+							   validate="false"
+							   method="get"
+							   mediatype="application/atom+xml"
+							   instance="i-atom-entries">
+					<xf:resource value="instance('i-controller')/feedURL"/>
-                <xf:message ev:event="xforms-submit-done" level="ephemeral">feed update successful</xf:message>
+					<xf:message ev:event="xforms-submit-done" level="ephemeral">feed update successful</xf:message>
-                <xf:message ev:event="xforms-submit-error">feed update failed</xf:message>
+					<xf:message ev:event="xforms-submit-error" level="modal">feed update failed</xf:message>
-            </xf:submission>
+				</xf:submission>
 
-        <xf:action ev:event="xforms-ready">
-            <xf:send submission="s-get-data"/>
-        </xf:action>
+				<xf:action ev:event="xforms-ready">
+					<xf:send submission="s-get-data"/>
+				</xf:action>
-
-    </xf:model>
+			</xf:model>
-</head>
-
-<body class="tundra" style="overflow:hidden;">
-
+		</div>
-    <xf:group id="repeatOuter" appearance="full">
+		<xf:group id="repeatOuter" appearance="full">
-        <xf:label class="heading">Atom Feed Reader</xf:label>
-
-        <div id="compactDescription">
-            Simple Display for ATOM Resources.<br/>
+			<div class="Section">
+				<div class="Headline">Atom Feed Reader</div>
+				Simple Display for ATOM Resources.
-            Shows Meta Data and lists all entries of an ATOM Resource.
-        </div>
+				Shows Meta Data and lists all entries of an ATOM Resource.
+			</div>
-        <xf:trigger id="reload">
+
+			<div class="Section">
+				<div class="Sample">
+					<xf:select1 appearance="minimal" ref="feedURL">
+						<xf:label>FeedURL:</xf:label>
+						<xf:choices>
+							<xf:item selected="True">
+								<xf:label>Golem</xf:label>
+								<xf:value>http://rss.golem.de/rss.php?feed=ATOM1.0</xf:value>
+							</xf:item>
+							<xf:item selected="False">
+								<xf:label>heise Online</xf:label>
+								<xf:value>http://www.heise.de/newsticker/heise-atom.xml</xf:value>
+							</xf:item>
+							<xf:item selected="False">
+								<xf:label>heise Security</xf:label>
+								<xf:value>http://www.heise.de/security/news/news-atom.xml</xf:value>
+							</xf:item>
+							<xf:item selected="False">
+								<xf:label>heise mobil</xf:label>
+								<xf:value>http://www.heise.de/mobil/newsticker/heise-atom.xml</xf:value>
+							</xf:item>
+							<xf:item selected="False">
+								<xf:label>heise Open</xf:label>
+								<xf:value>http://www.heise.de/open/news/news-atom.xml</xf:value>
+							</xf:item>
+							<xf:item selected="False">
+								<xf:label>iX</xf:label>
+								<xf:value>http://www.heise.de/ix/news/news-atom.xml</xf:value>
+							</xf:item>
+						</xf:choices>
+						<xf:action ev:event="xforms-value-changed">
+							<xf:send submission="s-get-data"/>
+						</xf:action>
+					</xf:select1>
+
+
+					<xf:trigger class="atomReload">
-            <xf:label></xf:label>
-            <xf:hint>Click to reload this feed</xf:hint>
-            <xf:action>
-                <xf:send submission="s-get-data"/>
-            </xf:action>
-        </xf:trigger>
-        <br/>
+						<xf:label></xf:label>
+						<xf:hint>Click to reload this feed</xf:hint>
+						<xf:action>
+							<xf:send submission="s-get-data"/>
+						</xf:action>
+					</xf:trigger>
+					<br/>
 
-        <xf:group id="meta-data" appearance="full">
+					<xf:group appearance="full" class="atomData">
-            <xf:label></xf:label>
+						<xf:label></xf:label>
-            <xf:output ref="instance('i-atom-entries')/atom:title" class="subtitle">
+						<xf:output ref="instance('i-atom-entries')/atom:title" class="atomSubtitle">
-                <xf:label>Feed: </xf:label>
+							<xf:label>Feed:</xf:label>
-            </xf:output>
-            <br/>
+						</xf:output>
+						<br/>
-            <xf:output ref="instance('i-atom-entries')/atom:subtitle" class="subtitle">
+						<xf:output ref="instance('i-atom-entries')/atom:subtitle" class="atomSubtitle">
-                <xf:label>Subtitle: </xf:label>
+							<xf:label>Subtitle:</xf:label>
-            </xf:output>
-            <br/>
+						</xf:output>
+						<br/>
-            <xf:output ref="instance('i-atom-entries')/atom:updated" class="subtitle">
+						<xf:output ref="instance('i-atom-entries')/atom:updated" class="atomSubtitle">
-                <xf:label>Last update: </xf:label>
+							<xf:label>Last update:</xf:label>
-            </xf:output>
-        </xf:group>
-        <!--Repeats the data-->
-        <xf:group id="whatever">
-            <xf:repeat nodeset="instance('i-atom-entries')/atom:entry" appearance="full">
-                <xf:group appearance="full">
+						</xf:output>
+					</xf:group>
+					<!--Repeats the data-->
+					<xf:group id="whatever">
+						<xf:repeat nodeset="instance('i-atom-entries')/atom:entry" appearance="full">
+							<xf:group appearance="full">
-                    <xf:output ref="atom:title" class="subtitle">
+								<xf:output ref="atom:title" class="atomSubtitle">
-                        <xf:label/>
-                    </xf:output>
-                    <xf:group appearance="ca:horizontalTable">
-                        <xf:label/>
+									<xf:label/>
+								</xf:output>
+								<xf:group appearance="ca:horizontalTable">
+									<xf:label/>
-                        <xf:output value="substring(atom:updated,1,10)" class="date">
+									<xf:output value="substring(atom:updated,1,10)" class="atomPadding">
-                            <xf:label/>
-                        </xf:output>
+										<xf:label/>
+									</xf:output>
-                        <xf:output value="substring(atom:updated,12,8)" class="time">
+									<xf:output value="substring(atom:updated,12,8)" class="atomPadding">
-                            <xf:label/>
-                        </xf:output>
-                    </xf:group>
-                    <xf:output class="summary" ref="atom:summary" mediatype="text/html">
-                        <xf:label/>
-                    </xf:output>                    
+										<xf:label/>
+									</xf:output>
+								</xf:group>
+								<xf:output class="summary" ref="atom:summary" mediatype="text/html">
+									<xf:label/>
+								</xf:output>
-                    <xf:trigger  class="articleLink" appearance="minimal">
+								<xf:trigger class="atomArticleLink" appearance="minimal">
-                        <xf:label>full article</xf:label>
+									<xf:label>full article</xf:label>
-                        <xf:load ref="atom:link/@href" show="new"/>
+									<xf:load show="new">
+										<xf:resource value="atom:link/@href"/>
+									</xf:load>
-                    </xf:trigger>
-                </xf:group>
-            </xf:repeat>
-        </xf:group>
+								</xf:trigger>
+							</xf:group>
+						</xf:repeat>
+					</xf:group>
+				</div>
+			</div>
-    </xf:group>
+		</xf:group>
+	</div>
 </body>
 </html>
Index: src/main/xforms/incubator/address2.xml
===================================================================
--- src/main/xforms/incubator/address2.xml	(revision 76)
+++ src/main/xforms/incubator/address2.xml	Thu Feb 18 16:33:58 CET 2010
@@ -1,111 +1,125 @@
 <?xml version="1.0" encoding="ISO-8859-1"?>
 <div
-      xmlns="http://www.w3.org/1999/xhtml"
-      xmlns:xf="http://www.w3.org/2002/xforms"
-      xmlns:ev="http://www.w3.org/2001/xml-events"
+        xmlns="http://www.w3.org/1999/xhtml"
+        xmlns:xf="http://www.w3.org/2002/xforms"
+        xmlns:ev="http://www.w3.org/2001/xml-events"
         xmlns:bffn="java:de.betterform.xml.xforms.xpath.BetterFormXPathFunctions">
-        <xf:model id="address" ev:event="xforms-revalidate" ev:defaultAction="cancel">
+    <xf:model id="address" ev:event="xforms-revalidate" ev:defaultAction="cancel">
 
 
-            <xf:instance id="default">
-                <address xmlns="">
+        <xf:instance id="default">
+            <address xmlns="">
+                <surname/>
+                <name/>
-                    <street/>
-                    <city/>
-                    <zip>123456</zip>
-                    <country>germany</country>
-                </address>
-            </xf:instance>
+                <street/>
+                <city/>
+                <zip>123456</zip>
+                <country>germany</country>
+            </address>
+        </xf:instance>
-            <xf:bind nodeset="zip"  type="integer"/>
+        <xf:bind nodeset="zip" type="integer"/>
 
-            <xf:submission id="s-load-from-master"
-                           resource="model:master#instance('contact')/data/address[index('addresses')]"
-                           replace="instance"
+        <xf:submission id="s-load-from-master"
+                       resource="model:master#instance('contact')/data/address[index('addresses')]"
+                       replace="instance"
-                           method="get"/>
+                       method="get">
+            <xf:message ev:event="xforms-submit-done" level="ephemeral">Subform has updated itself.</xf:message>
+        </xf:submission>
+
-            <xf:submission id="s-update-master"
-                           resource="model:master#instance('contact')/data/address[index('addresses')]"
-                           replace="none"
+        <xf:submission id="s-update-master"
+                       resource="model:master#instance('contact')/data/address[index('addresses')]"
+                       replace="none"
-                           method="post"/>
+                       method="post">
+            <xf:message ev:event="xforms-submit-done" level="ephemeral">Subform has updated Master.</xf:message>
+        </xf:submission>
 
-            <xf:send ev:event="xforms-ready" submission="s-load-from-master"/>
-        </xf:model>
-        <xf:group appearance="full">
-            <xf:action ev:event="betterform-variable-changed" ev:observer="contact">
-                <xf:message>it worked</xf:message>
-            </xf:action>
+        <xf:send ev:event="xforms-ready" submission="s-load-from-master"/>
+    </xf:model>
+    <xf:group appearance="full">
+        <xf:action ev:event="betterform-variable-changed" ev:observer="contact">
+            <xf:message>it worked</xf:message>
+        </xf:action>
-            <xf:label>Address Form 2</xf:label>
+        <xf:label>Addressdialog Subfrom 2</xf:label>
 
+
+        <xf:input ref="surname">
+            <xf:label id="surname">Surname</xf:label>
+        </xf:input>
+        <xf:input ref="name">
+            <xf:label id="name">Name</xf:label>
+        </xf:input>
+        <xf:input ref="street">
+            <xf:label id="street">Street</xf:label>
+        </xf:input>
+        <xf:input ref="city">
+            <xf:label id="city">City</xf:label>
+        </xf:input>
+        <xf:input ref="zip">
+            <xf:label id="zip">Zip</xf:label>
+            <xf:alert>Please put your ZIP code here</xf:alert>
+        </xf:input>
-            <xf:select1  appearance="minimal" ref="country">
+        <xf:select1 appearance="minimal" ref="country">
-                <xf:label id="country-lbl">Country</xf:label>
-                <xf:choices id="C-38">
+            <xf:label id="country">Country</xf:label>
+            <xf:choices>
-                    <xf:item selected="false">
-                        <xf:label>Albania</xf:label>
-                        <xf:value>Albania</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>Belgium</xf:label>
-                        <xf:value>Belgium</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>Cuba</xf:label>
-                        <xf:value>Cuba</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>Denmark</xf:label>
-                        <xf:value>Denmark</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>Estonia</xf:label>
-                        <xf:value>Estonia</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>France</xf:label>
-                        <xf:value>France</xf:value>
-                    </xf:item>
-                    <xf:item selected="true">
-                        <xf:label>Germany</xf:label>
-                        <xf:value>Germany</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>Gibraltar</xf:label>
-                        <xf:value>Gibraltar</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>Hungary</xf:label>
-                        <xf:value>Hungary</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>Italy</xf:label>
-                        <xf:value>Italy</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>Japan</xf:label>
-                        <xf:value>Japan</xf:value>
-                    </xf:item>
-                    <xf:item selected="false">
-                        <xf:label>Spain</xf:label>
-                        <xf:value>Spain</xf:value>
-                    </xf:item>
-                </xf:choices>
-            </xf:select1>
+                <xf:item selected="false">
+                    <xf:label>Albania</xf:label>
+                    <xf:value>Albania</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>Belgium</xf:label>
+                    <xf:value>Belgium</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>Cuba</xf:label>
+                    <xf:value>Cuba</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>Denmark</xf:label>
+                    <xf:value>Denmark</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>Estonia</xf:label>
+                    <xf:value>Estonia</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>France</xf:label>
+                    <xf:value>France</xf:value>
+                </xf:item>
+                <xf:item selected="true">
+                    <xf:label>Germany</xf:label>
+                    <xf:value>Germany</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>Gibraltar</xf:label>
+                    <xf:value>Gibraltar</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>Hungary</xf:label>
+                    <xf:value>Hungary</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>Italy</xf:label>
+                    <xf:value>Italy</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>Japan</xf:label>
+                    <xf:value>Japan</xf:value>
+                </xf:item>
+                <xf:item selected="false">
+                    <xf:label>Spain</xf:label>
+                    <xf:value>Spain</xf:value>
+                </xf:item>
+            </xf:choices>
+        </xf:select1>
 
-            <xf:input ref="city">
-                <xf:label id="city-lbl">City</xf:label>
-            </xf:input>
-            <xf:input ref="zip">
-                <xf:label id="zip-lbl">Zip</xf:label>
-                <xf:alert>Please put your ZIP code here</xf:alert>
-            </xf:input>
-            <xf:input ref="street">
-                <xf:label id="street">Street</xf:label>
-            </xf:input>
 
-            <xf:trigger>
-                <xf:label>load data from master</xf:label>
-                <xf:send submission="s-load-from-master"/>
-            </xf:trigger>
-            <xf:trigger>
-                <xf:label>update master</xf:label>
-                <xf:send submission="s-update-master"/>
-            </xf:trigger>
-        </xf:group>
+        <xf:trigger>
+            <xf:label>load data from master</xf:label>
+            <xf:send submission="s-load-from-master"/>
+        </xf:trigger>
+        <xf:trigger>
+            <xf:label>update master</xf:label>
+            <xf:send submission="s-update-master"/>
+        </xf:trigger>
+    </xf:group>
 </div>
Index: src/main/xforms/incubator/orderlist.xhtml
===================================================================
--- src/main/xforms/incubator/orderlist.xhtml	(revision 9)
+++ src/main/xforms/incubator/orderlist.xhtml	Fri Feb 19 20:38:07 CET 2010
@@ -1,135 +1,217 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!-- $Id: orderlist.xhtml 2521 2007-03-26 09:05:09Z joernt $ -->
 <html xmlns="http://www.w3.org/1999/xhtml"
-    xmlns:xf="http://www.w3.org/2002/xforms"
-    xmlns:ev="http://www.w3.org/2001/xml-events">
-    <head>
-        <title>Order List</title>
+	  xmlns:xf="http://www.w3.org/2002/xforms"
+	  xmlns:ev="http://www.w3.org/2001/xml-events">
+<head>
+	<title>Order List</title>
-        <style type="text/css">
-            .text-right {text-align: right;}
-            .repeat-header *{background:darkkhaki;}
-            .repeat-header label{padding:5px;font-weight:bold;}
-            fieldset {width:400px;}
-            fieldset fieldset{padding:0;}
-            fieldset fieldset table{margin-left:auto;margin-right:auto;text-align:center;}
-            .repeat-header{background:cadetblue;}
-            .repeat-header label{padding:5px;font-weight:bold;}
-/*
-            .col-1 label{width:50px;}
-            .col-2 {text-align:center;}
-            .col-2 .value{width:30px;text-align:center;}
-            .col-2 label{width:50px;}
+	<meta name="author" content="joernt at users.sourceforge.net"/>
+	<meta name="author" content="lars.windauer at chiba-project.org"/>
+	<meta name="author" content="tobias.krebs at betterform.de"/>
+	<meta name="author" content="fabian.otto at betterform.de"/>
 
-            .col-3 label{width:70px;}
-            .col-4 label{width:70px;}
-*/
+	<meta name="description" content="XForms demo of a simple order list"/>
 
-            #total {background:palegoldenrod;display:block;border-top:solid thin rgb( 188, 216, 132 );text-align:right;margin-top:20px;}
-            #total label{font-size:16pt;padding-top:5px;text-align:left;margin-left:30px;}
-            #total-value{display:block;padding:5px;font-size:16pt;padding-top:5px;white-space:nowrap;margin-right:30px;}
+	<link rel="stylesheet" type="text/css" href="../../styles/bf.css"/>
+	<link rel="stylesheet" type="text/css" href="../../resources/styles/reference.css"/>
+	<link rel="stylesheet" type="text/css" href="../../resources/styles/demo.css"/>
 
-
-        </style>
+</head>
+<body class="soria" style="margin:30px;">
+<div id="xforms">
+<div style="display:none">
-        <xf:model>
-            <xf:instance xmlns="">
-                <order>
+	<xf:model>
+		<xf:instance xmlns="">
+			<order>
-                    <item amount="1" price="2.50" sum="0">article_1</item>
-                    <item amount="1" price="5.00" sum="0">article_2</item>
-                    <item amount="1" price="9.95" sum="">article_3</item>
-<!--
-                    <item amount="1" price="15.37" sum="">article_4</item>
-                    <item amount="1" price="12.25" sum="">article_5</item>
--->
-                    <item amount="1" price="0" sum="0"/>
+				<item amount="1" price="1.50" sum="0" tax="0">pencil</item>
+				<item amount="1" price="2.50" sum="" tax="0">ballpoint_pen</item>
+				<item amount="1" price="50.00" sum="0" tax="0">fountain-pen</item>
+				<item amount="1" price="0" sum="0" tax="0"/>
-                </order>
-            </xf:instance>
+			</order>
+		</xf:instance>
 
+		<xf:instance id="shippingAndTaxes">
+			<order xmlns=""/>
+		</xf:instance>
+
+		<xf:instance id="countries">
+			<contries xmlns="">
+				<country>
+					<name>Albania</name>
+					<shipping>12.50</shipping>
+					<tax>10.0</tax>
+				</country>
+				<country>
+					<name>Belgium</name>
+					<shipping>8.00</shipping>
+					<tax>20.0</tax>
+				</country>
+				<country>
+					<name>Cuba</name>
+					<shipping>30.00</shipping>
+					<tax>0.0</tax>
+				</country>
+				<country>
+					<name>Denmark</name>
+					<shipping>10.00</shipping>
+					<tax>21.0</tax>
+				</country>
+				<country>
+					<name>Estonia</name>
+					<shipping>15.00</shipping>
+					<tax>7.0</tax>
+				</country>
+				<country>
+					<name>France</name>
+					<shipping>8.50</shipping>
+					<tax>16.0</tax>
+				</country>
+				<country>
+					<name>Germany</name>
+					<shipping>4.40</shipping>
+					<tax>19.0</tax>
+				</country>
+				<country>
+					<name>Gibraltar</name>
+					<shipping>13.25</shipping>
+					<tax>12.5</tax>
+				</country>
+				<country>
+					<name>Hungary</name>
+					<shipping>14.00</shipping>
+					<tax>6.2</tax>
+				</country>
+				<country>
+					<name>Italy</name>
+					<shipping>8.00</shipping>
+					<tax>16.0</tax>
+				</country>
+				<country>
+					<name>Japan</name>
+					<shipping>40.00</shipping>
+					<tax>21.5</tax>
+				</country>
+				<country>
+					<name>Spain</name>
+					<shipping>9.00</shipping>
+					<tax>14.0</tax>
+				</country>
+			</contries>
+		</xf:instance>
+
-            <xf:bind id="items" nodeset="item[position()!=last()]">
-                <xf:bind id="article" nodeset="." required="true()"/>
-                <xf:bind id="amount" nodeset="@amount" type="integer" required="true()" constraint=". &lt; 100"/>
-                <xf:bind id="price" nodeset="@price" type="float"/>
+		<xf:bind id="items" nodeset="item[position()!=last()]">
+			<xf:bind id="article" nodeset="." required="true()"/>
+			<xf:bind id="amount" nodeset="@amount" type="integer" required="true()" constraint=". &lt; 100"/>
+			<xf:bind id="price" nodeset="@price" type="float"/>
+			<xf:bind id="tax" nodeset="@tax" type="float"/>
-                <xf:bind id="sum" nodeset="@sum" calculate="../@amount * ../@price" type="float" />
+			<xf:bind id="sum" nodeset="@sum" calculate="../@amount * ../@price" type="float"/>
-            </xf:bind>
-        </xf:model>
+		</xf:bind>
+	</xf:model>
-    </head>
-    <body>
+</div>
+<div class="Section" dojotype="dijit.layout.ContentPane">
-        <xf:group appearance="full">
+	<xf:group appearance="full">
-            <xf:label>A simple calculated Order List with slightly formatted output.</xf:label>
+		<div class="Headline">Order List:</div>
 
+		<div class="description">
+			A simple calculated Order List with slightly formatted output.
+		</div>
+
+		<div class="orderListCountrySelect">
+			<xf:select1 appearance="minimal" ref="instance('shippingAndTaxes')">
+				<xf:label id="country">Ship to:</xf:label>
+				<xf:itemset nodeset="instance('countries')/country">
+					<xf:label ref="name"/>
+					<xf:copy ref="."/>
+				</xf:itemset>
+			</xf:select1>
+		</div>
+
+		<br/>
+		<xf:input ref="instance('shippingAndTaxes')/country/name">
+			<xf:label>Name:</xf:label>
+		</xf:input>
+		<br/>
+
-            <xf:group appearance="minimal">
+		<xf:group appearance="minimal">
-                <xf:repeat id="repeatid" bind="items" appearance="compact">
+			<xf:repeat id="repeatid" bind="items" appearance="compact" class="orderListRepeat">
-                    <xf:select1 bind="article">
+				<xf:select1 bind="article">
-                        <xf:label>Article</xf:label>
+					<xf:label class="contactsRepeatHeader">Article</xf:label>
-                        <xf:choices>
-                            <xf:item>
+					<xf:choices>
+						<xf:item>
-                                <xf:label>Article 1</xf:label>
-                                <xf:value>article_1</xf:value>
-                                <xf:setvalue ev:event="xforms-select" bind="price">2.50</xf:setvalue>
+							<xf:label>Eraser</xf:label>
+							<xf:value>eraser</xf:value>
+							<xf:setvalue ev:event="xforms-select" bind="price">0.50</xf:setvalue>
-                            </xf:item>
-                            <xf:item>
+						</xf:item>
+						<xf:item>
-                                <xf:label>Article 2</xf:label>
-                                <xf:value>article_2</xf:value>
-                                <xf:setvalue ev:event="xforms-select" bind="price">5.00</xf:setvalue>
+							<xf:label>Pencil</xf:label>
+							<xf:value>pencil</xf:value>
+							<xf:setvalue ev:event="xforms-select" bind="price">1.50</xf:setvalue>
-                            </xf:item>
-                            <xf:item>
+						</xf:item>
+						<xf:item>
-                                <xf:label>Article 3</xf:label>
-                                <xf:value>article_3</xf:value>
-                                <xf:setvalue ev:event="xforms-select" bind="price">9.95</xf:setvalue>
+							<xf:label>Ballpoint pen</xf:label>
+							<xf:value>ballpoint_pen</xf:value>
+							<xf:setvalue ev:event="xforms-select" bind="price">2.49</xf:setvalue>
-                            </xf:item>
-                            <xf:item>
+						</xf:item>
+						<xf:item>
-                                <xf:label>Article 4</xf:label>
-                                <xf:value>article_4</xf:value>
-                                <xf:setvalue ev:event="xforms-select" bind="price">15.37</xf:setvalue>
+							<xf:label>Spiral notebook (leather binding)</xf:label>
+							<xf:value>spiral_notebook</xf:value>
+							<xf:setvalue ev:event="xforms-select" bind="price">3.99</xf:setvalue>
-                            </xf:item>
-                            <xf:item>
+						</xf:item>
+						<xf:item>
-                                <xf:label>Article 5</xf:label>
-                                <xf:value>article_5</xf:value>
-                                <xf:setvalue ev:event="xforms-select" bind="price">12.25</xf:setvalue>
+							<xf:label>Fountain pen</xf:label>
+							<xf:value>fountain-pen</xf:value>
+							<xf:setvalue ev:event="xforms-select" bind="price">50.00</xf:setvalue>
-                            </xf:item>
-                            <xf:item>
+						</xf:item>
+						<xf:item>
-                                <xf:label>Article 6</xf:label>
-                                <xf:value>article_6</xf:value>
-                                <xf:setvalue ev:event="xforms-select" bind="price">47.11</xf:setvalue>
+							<xf:label>Shredder</xf:label>
+							<xf:value>shredder</xf:value>
+							<xf:setvalue ev:event="xforms-select" bind="price">69.98</xf:setvalue>
-                            </xf:item>
-                        </xf:choices>
-                    </xf:select1>
-                    <xf:input bind="amount" incremental="true">
+						</xf:item>
+					</xf:choices>
+				</xf:select1>
+				<xf:input bind="amount" incremental="true">
-                        <xf:label>Amount</xf:label>
+					<xf:label class="contactsRepeatHeader">Amount</xf:label>
-                        <xf:alert>You can order a maximum of 99 items in one order</xf:alert>
-                    </xf:input>
-                    <xf:output value="format-number(@price, 'Û 0.00')" class="text-right">
+					<xf:alert>You can order a maximum of 99 items in one order</xf:alert>
+				</xf:input>
+				<xf:output value="format-number(@price, 'Û 0.00')" class="text-right">
-                        <xf:label>Price</xf:label>
+					<xf:label class="contactsRepeatHeader">Price</xf:label>
-                    </xf:output>
+				</xf:output>
+				<xf:output value="format-number(@tax, 'Û 0.00')" class="text-right">
+					<xf:label class="contactsRepeatHeader">Tax</xf:label>
+				</xf:output>
-                    <xf:output value="format-number(@sum, 'Û 0.00')" class="text-right">
+				<xf:output value="format-number(@sum, 'Û 0.00')" class="text-right">
-                        <xf:label>Sum</xf:label>
+					<xf:label class="contactsRepeatHeader">Sum</xf:label>
-                    </xf:output>
-                </xf:repeat>
+				</xf:output>
+			</xf:repeat>
 
-                <xf:output id="total" value="format-number(sum(/order/item/@sum), 'Û 0.00')">
+			<xf:output id="total" value="format-number(sum(/order/item/@sum), 'Û 0.00')">
-                    <xf:label>Total Price: </xf:label>
+				<xf:label class="contactsRepeatHeader">Total Price:</xf:label>
-                </xf:output>
-            </xf:group>
+			</xf:output>
+		</xf:group>
+	</xf:group>
 
-            <xf:group appearance="minimal">
-                <xf:trigger>
+	<xf:group appearance="minimal" class="orderListTriggerGroup">
+		<xf:trigger class="orderListAddTrigger">
-                    <xf:label>Add item</xf:label>
-                    <xf:action>
-                        <xf:insert bind="items" at="last()" position="after"/>
-                    </xf:action>
-                </xf:trigger>
+			<xf:label>Add item</xf:label>
+			<xf:action>
+				<xf:insert bind="items" at="last()" position="after"/>
+			</xf:action>
+		</xf:trigger>
-                <xf:trigger>
+		<xf:trigger class="orderListDelTrigger">>
-                    <xf:label>Delete item</xf:label>
-                    <xf:action>
-                        <xf:delete bind="items" at="index('repeatid')"/>
-                    </xf:action>
-                </xf:trigger>
+			<xf:label>Delete item</xf:label>
+			<xf:action>
+				<xf:delete bind="items" at="index('repeatid')"/>
+			</xf:action>
+		</xf:trigger>
-                <xf:trigger>
+		<xf:trigger class="orderListResetTrigger">>
-                    <xf:label>Reset</xf:label>
-                    <xf:action>
-                        <xf:reset/>
-                    </xf:action>
-                </xf:trigger>
-            </xf:group>
+			<xf:label>Reset</xf:label>
+			<xf:action>
+				<xf:reset/>
+			</xf:action>
+		</xf:trigger>
+	</xf:group>
-        </xf:group>
+
+</div>
+</div>
-    </body>
+</body>
 </html>
Index: src/main/resources/styles/reference.css
===================================================================
--- src/main/resources/styles/reference.css	(revision 9)
+++ src/main/resources/styles/reference.css	Fri Feb 19 15:46:25 CET 2010
@@ -92,7 +92,7 @@
     border:thin solid #cccccc;
 }
 .Sample {
-    background-color: #f0f0ff;
+    background-color: lightsteelblue;
     vertical-align: middle;
     padding: 20px 10px 20px 10px;
     overflow: auto;
Index: src/main/resources/styles/demo.css
===================================================================
--- src/main/resources/styles/demo.css	Fri Feb 19 19:33:46 CET 2010
+++ src/main/resources/styles/demo.css	Fri Feb 19 19:33:46 CET 2010
@@ -0,0 +1,218 @@
+/* Atom Demo */
+.atomReload {
+	height: 30px;
+	width: 30px;
+	padding: 10px;
+	float: right;
+}
+
+.atomReload button {
+	height: 30px;
+	width: 30px;
+	background: url(../../resources/images/reload.gif) no-repeat center;
+}
+
+.atomSubtitle .xfLabel {
+	font-weight: bold;
+	font-size: medium;
+	display: block;
+}
+
+.atomSubtitle .xfValue {
+	font-weight: bolder;
+	padding: 5px;
+	font-size: 12pt;
+}
+
+.atomData {
+	display: block;
+	padding: 20px;
+}
+
+.atomPadding .xfValue {
+	padding: 5px;
+}
+
+.atomArticleLink {
+	padding: 5px;
+}
+
+.atomArticleLink .xfValue {
+	padding-top: 5px;
+}
+
+/* Contacts */
+.contactsRepeatHeader {
+	margin-bottom: 5px;
+	font-weight: bold;
+}
+
+.contactsRepeat {
+	border: 1px solid black;
+	padding: 5px;
+	width: 100%;
+}
+
+.contactsRepeat .xfRepeatHeader {
+	padding: 5px;
+}
+
+.contactsRepeat .xfRepeatItem .xfValue {
+	padding: 3px;
+	display: block;
+}
+
+.contactsSubForm {
+	border: 1px solid black;
+	display: block;
+}
+
+.contactsSubForm .xfGroupLabel {
+	padding: 5px;
+	font-weight: bold;
+	font-size: 12pt;
+}
+
+.contactsSubForm .xfLabel {
+	padding: 3px 3px 3px 8px;
+	width: 7em;
+}
+
+.contactsFullGroup .xfLabel {
+	width: 7em;
+}
+
+.contactsTriggerGroup {
+	display: block;
+	text-align: center;
+	padding: 5px;
+}
+
+.contactsMasterTrigger {
+	display: inline;
+	text-align: center;
+	width: 200px;
+	padding-right: 15px;
+}
+
+.contactsMasterTrigger button {
+	background: url(../../resources/images/001_22.png) no-repeat 5px 5px;
+	display: block;
+	height: 30px;
+	line-height: 29px;
+	text-decoration: none;
+	width: 200px;
+	text-indent: 15px;
+}
+
+.contactsSubTrigger {
+	display: inline;
+	text-align: center;
+	width: 200px;
+	padding-right: 15px;
+}
+
+.contactsSubTrigger button {
+	background: url(../../resources/images/001_22.png) no-repeat 5px 5px;
+	display: block;
+	height: 30px;
+	line-height: 29px;
+	text-decoration: none;
+	width: 200px;
+	text-indent: 15px;
+
+}
+
+.contactsUpdateMasterTrigger {
+	display: inline;
+	text-align: center;
+	width: 200px;
+}
+
+.contactsUpdateMasterTrigger button {
+	background: url(../../resources/images/001_24.png) no-repeat 5px 5px;
+	display: block;
+	height: 30px;
+	line-height: 29px;
+	text-decoration: none;
+	width: 200px;
+	text-indent: 15px;
+}
+
+/* Order List */
+.orderListHeader {
+	margin-bottom: 5px;
+	font-weight: bold;
+}
+
+.orderListRepeat {
+	border: 1px solid black;
+	padding: 5px;
+	width: 100%;
+}
+
+.orderListRepeat .xfRepeatHeader {
+	padding: 5px;
+}
+
+.orderListRepeat .xfRepeatItem .xfValue {
+	padding: 3px;
+	display: block;
+}
+
+.orderListTriggerGroup {
+	display: block;
+	text-align: center;
+	padding: 5px;
+}
+
+.orderListAddTrigger {
+	display: inline;
+	text-align: center;
+	width: 200px;
+	padding-right: 15px;
+}
+
+.orderListAddTrigger button {
+	background: url(../../resources/images/001_01.png) no-repeat 5px 5px;
+	display: block;
+	height: 30px;
+	line-height: 29px;
+	text-decoration: none;
+	width: 200px;
+	text-indent: 15px;
+}
+
+.orderListDelTrigger {
+	display: inline;
+	text-align: center;
+	width: 200px;
+	padding-right: 15px;
+}
+
+.orderListDelTrigger button {
+	background: url(../../resources/images/001_02.png) no-repeat 5px 5px;
+	display: block;
+	height: 30px;
+	line-height: 29px;
+	text-decoration: none;
+	width: 200px;
+	text-indent: 15px;
+
+}
+
+.orderListResetTrigger {
+	display: inline;
+	text-align: center;
+	width: 200px;
+}
+
+.orderListResetTrigger button {
+	background: url(../../resources/images/001_39.png) no-repeat 5px 5px;
+	display: block;
+	height: 30px;
+	line-height: 29px;
+	text-decoration: none;
+	width: 200px;
+	text-indent: 15px;
+}
Index: web/src/main/webapp/demo.xhtml
===================================================================
--- web/src/main/webapp/demo.xhtml	(revision 144)
+++ web/src/main/webapp/demo.xhtml	Fri Feb 19 19:18:22 CET 2010
@@ -1,6 +1,7 @@
 <html xmlns="http://www.w3.org/1999/xhtml"
       xmlns:xf="http://www.w3.org/2002/xforms"
       xmlns:ev="http://www.w3.org/2001/xml-events"
+	  xmlns:atom="http://www.w3.org/2005/Atom"
       xml:lang="en" lang="en">
 	<head>
 		<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
@@ -16,6 +17,7 @@
 		<link rel="ICON" href="images/betterform.ico" type="image/x-icon"/>
 		<link rel="stylesheet" type="text/css" href="styles/bf.css"/>
         <link rel="stylesheet" type="text/css" href="resources/styles/reference.css" />
+		<link rel="stylesheet" type="text/css" href="resources/styles/demo.css" />		
 
         <style type="text/css">
             #demoTarget{
@@ -35,9 +37,41 @@
             }
             #demoTrigger{display:none;}
         </style>
+	
-        <script type="text/javascript">
+		<script type="text/javascript">
+        dojo.require("dojox.highlight");
+        dojo.require("dojox.highlight.languages.xml");
+        //dojo.require("dojox.highlight.languages._static");
+        //dojo.require("dojox.highlight.languages._dynamic");
+        //dojo.require("dojox.highlight.languages._www");
+        dojo.addOnLoad(function() {
+            dojo.query("code").forEach(dojox.highlight.init);
+        });
+
+        dojo.require("dijit.form.Button");
-            dojo.require("dijit.TitlePane");
+        dojo.require("dijit.TitlePane");
+        dojo.require('dijit.layout.ContentPane');
+        <!-- todo: resolve issue with dojo.requires appearing in subforms. These must be present in master at the
+        moment but this shouldn't be necessary .-->
+        dojo.require("betterform.ui.select.OptGroup");
+        dojo.require("betterform.ui.select1.RadioItemset");
+        dojo.require("betterform.ui.select.CheckBoxItemset");
+		dojo.require("betterform.ui.util");
+        dojo.require("betterform.ui.container.Switch");
+        dojo.require("betterform.ui.container.Repeat");
+        dojo.require("betterform.ui.container.RepeatItem");
+		dojo.require("betterform.ui.container.TabSwitch");
+		dojo.require("betterform.ui.container.Group");
+
+        function addToDocument(id){
+           var model = dojo.query(".xfModel", dojo.doc)[0];
+           dijit.byId(dojo.attr(model, "id")).getInstanceAsString(id,
+               function(data){
+                   dojo.byId("instanceData").innerHTML=data;
+               });
+        }
+
-        </script>
+    </script>
 
 
 	</head>
@@ -56,7 +90,14 @@
             <xf:trigger id="demoTrigger">
                 <xf:label>run demo</xf:label>
                 <xf:load show="embed" targetid="demoTarget">
-                    <xf:resource value="'forms/reference/Input.xhtml#xforms'"/>
+                    <!-- <xf:resource value="'forms/reference/Input.xhtml#xforms'"/> -->
+					<!-- <xf:resource value="'forms/reference/Group.xhtml#xforms'"/>
+					<xf:resource value="'forms/reference/Repeat.xhtml#xforms'"/>
+					<xf:resource value="'forms/demo/Atom.xhtml#xforms'"/>
+					<xf:resource value="'forms/demo/Contacts.xhtml#xforms'"/> -->
+					<!-- <xf:resource value="'forms/incubator/XFormsFeatureExplorer.xhtml#xforms'"/> -->
+					<xf:resource value="'forms/incubator/orderlist.xhtml#xforms'"/>
+
                 </xf:load>
             </xf:trigger>
 
Index: src/main/xforms/incubator/XFormsFeatureExplorer.xhtml
===================================================================
--- src/main/xforms/incubator/XFormsFeatureExplorer.xhtml	(revision 9)
+++ src/main/xforms/incubator/XFormsFeatureExplorer.xhtml	Fri Feb 19 18:02:34 CET 2010
@@ -54,6 +54,7 @@
     </style>
 </head>
 <body class="soria" style="width:100%;height:100%;margin:0;padding:0;overflow:hidden;">
+<div id="xforms">
     <div dojotype="dijit.layout.ContentPane" id="widgetsContainer">
         <div style="display:none">
             <xf:model>
@@ -129,7 +130,7 @@
                                         <xf:label ref="./@name"/>
                                         <xf:action>
                                             <xf:load show="embed" targetid="controlMount">
-                                                <xf:resource value="'../reference/Input.xhtml#xforms'"/>
+                                                <xf:resource value="concat(bffn:appContext('contextroot'),'/forms/reference/Input.xhtml#xforms')"/>
                                             </xf:load>
                                             <script type="text/javascript">
                                                 dojo.attr(dijit.byId('controlPane'), "open", "false");
@@ -147,7 +148,7 @@
                                         <xf:label ref="./@name"/>
                                         <xf:action>
                                             <xf:load show="embed" targetid="controlMount">
-                                                <xf:resource value="'../reference/Output.xhtml#xforms'"/>
+                                                <xf:resource value="concat(bffn:appContext('contextroot'),'/forms/reference/Output.xhtml#xforms')"/>
                                             </xf:load>
                                             <script type="text/javascript">
                                                 dojo.attr(dijit.byId('controlPane'), "open", "false");
@@ -165,7 +166,7 @@
                                         <xf:label ref="./@name"/>
                                         <xf:action>
                                             <xf:load show="embed" targetid="controlMount">
-                                                <xf:resource value="'../reference/Range.xhtml#xforms'"/>
+                                                <xf:resource value="concat(bffn:appContext('contextroot'),'/forms/reference/Range.xhtml#xforms')"/>
                                             </xf:load>
                                             <script type="text/javascript">
                                                 dojo.attr(dijit.byId('controlPane'), "open", "false");
@@ -183,7 +184,7 @@
                                         <xf:label ref="./@name"/>
                                         <xf:action>
                                             <xf:load show="embed" targetid="controlMount">
-                                                <xf:resource value="'../reference/Secret.xhtml#xforms'"/>
+                                                <xf:resource value="concat(bffn:appContext('contextroot'),'/forms/reference/Secret.xhtml#xforms')"/>
                                             </xf:load>
                                             <script type="text/javascript">
                                                 dojo.attr(dijit.byId('controlPane'), "open", "false");
@@ -201,7 +202,7 @@
                                         <xf:label ref="./@name"/>
                                         <xf:action>
                                             <xf:load show="embed" targetid="controlMount">
-                                                <xf:resource value="'../reference/Select.xhtml#xforms'"/>
+                                                <xf:resource value="concat(bffn:appContext('contextroot'),'/forms/reference/Select.xhtml#xforms')"/>
                                             </xf:load>
                                             <script type="text/javascript">
                                                 dojo.attr(dijit.byId('controlPane'), "open", "false");
@@ -219,7 +220,8 @@
                                         <xf:label ref="./@name"/>
                                         <xf:action>
                                             <xf:load show="embed" targetid="controlMount">
+                                                <!-- <xf:resource value="concat(bffn:appContext('contextroot'),'/forms/reference/Select1.xhtml#xforms')"/> -->
-                                                <xf:resource value="'../reference/Select1.xhtml#xforms'"/>
+												<xf:resource value="'../reference/Select1.xhtml#xforms'"/>
                                             </xf:load>
                                             <script type="text/javascript">
                                                 dojo.attr(dijit.byId('controlPane'), "open", "false");
@@ -237,7 +239,7 @@
                                         <xf:label ref="./@name"/>
                                         <xf:action>
                                             <xf:load show="embed" targetid="controlMount">
-                                                <xf:resource value="'../reference/Textarea.xhtml#xforms'"/>
+                                                <xf:resource value="concat(bffn:appContext('contextroot'),'/forms/reference/Textarea.xhtml#xforms')"/>
                                             </xf:load>
                                             <script type="text/javascript">
                                                 dojo.attr(dijit.byId('controlPane'), "open", "false");
@@ -255,7 +257,7 @@
                                         <xf:label ref="./@name"/>
                                         <xf:action>
                                             <xf:load show="embed" targetid="controlMount">
-                                                <xf:resource value="'../reference/Trigger.xhtml#xforms'"/>
+                                                <xf:resource value="concat(bffn:appContext('contextroot'),'/forms/reference/Trigger.xhtml#xforms')"/>
                                             </xf:load>
                                             <script type="text/javascript">
                                                 dojo.attr(dijit.byId('controlPane'), "open", "false");
@@ -273,7 +275,7 @@
                                         <xf:label ref="./@name"/>
                                         <xf:action>
                                             <xf:load show="embed" targetid="controlMount">
-                                                <xf:resource value="'../reference/Upload.xhtml#xforms'"/>
+                                                <xf:resource value="concat(bffn:appContext('contextroot'),'/forms/reference/Upload.xhtml#xforms')"/>
                                             </xf:load>
                                             <script type="text/javascript">
                                                 dojo.attr(dijit.byId('controlPane'), "open", "false");
@@ -306,7 +308,7 @@
                                         <xf:label ref="./@name"/>
                                         <xf:action>
                                             <xf:load show="embed" targetid="containerMount">
-                                                <xf:resource value="'../reference/Group.xhtml#xforms'"/>
+                                                <xf:resource value="concat(bffn:appContext('contextroot'),'/forms/reference/Group.xhtml#xforms')"/>
                                             </xf:load>
                                             <script type="text/javascript">
                                                 dojo.attr(dijit.byId('containerPane'), "open", "false");
@@ -324,7 +326,7 @@
                                         <xf:label ref="./@name"/>
                                         <xf:action>
                                             <xf:load show="embed" targetid="containerMount">
-                                                <xf:resource value="'../reference/Repeat.xhtml#xforms'"/>
+                                                <xf:resource value="concat(bffn:appContext('contextroot'),'/forms/reference/Repeat.xhtml#xforms')"/>
                                             </xf:load>
                                             <script type="text/javascript">
                                                 dojo.attr(dijit.byId('containerPane'), "open", "false");
@@ -342,7 +344,7 @@
                                         <xf:label ref="./@name"/>
                                         <xf:action>
                                             <xf:load show="embed" targetid="containerMount">
-                                                <xf:resource value="'../reference/Switch.xhtml#xforms'"/>
+                                                <xf:resource value="concat(bffn:appContext('contextroot'),'/forms/reference/Switch.xhtml#xforms')"/>
                                             </xf:load>
                                             <script type="text/javascript">
                                                 dojo.attr(dijit.byId('containerPane'), "open", "false");
@@ -364,5 +366,7 @@
 
         </xf:switch>
     </div>
+</div>
 </body>
 </html>
+                                                                                                                                               
Index: src/main/xforms/reference/Select1.xhtml
===================================================================
--- src/main/xforms/reference/Select1.xhtml	(revision 125)
+++ src/main/xforms/reference/Select1.xhtml	Fri Feb 19 15:43:48 CET 2010
@@ -49,8 +49,8 @@
         </script>
 
     </head>
-    <body class="soria" style="margin:30px;">
-
+    <body class="soria" style="margin:30px;overflow:scroll;">
+	    <div style="height:800px"/>
         <div id="xforms">
             <div style="display:none;">
                 <xf:model id="model-1">
