--- dojo-release-1.7.2-src-original/util/build/transforms/writeOptimized.js	2012-02-08 10:55:31.000000000 +0100
+++ dojo-release-1.7.2-src-modified/util/build/transforms/writeOptimized.js	2012-03-08 15:33:05.575008200 +0100
@@ -4,6 +4,17 @@
 	// default to a no-op
 	var compile = function(){};
 
+	var stripConsoleRe = 0;
+	if(bc.stripConsole){
+		var consoleMethods = "assert|count|debug|dir|dirxml|group|groupEnd|info|profile|profileEnd|time|timeEnd|trace|log";
+		if(bc.stripConsole == "warn"){
+			consoleMethods += "|warn";
+		}else if(bc.stripConsole == "all"){
+			consoleMethods += "|warn|error";
+		}	
+		stripConsoleRe= new RegExp("console\\.(" + consoleMethods + ")\\s*\\(", "g");
+	}
+
 	if(has("host-rhino") && (bc.optimize || bc.layerOptimize)){
 		function sscompile(text, dest, optimizeSwitch, copyright){
 			// decode the optimize switch
@@ -256,17 +267,6 @@
 			}
 		});
 
-		var stripConsoleRe= 0;
-		if(bc.stripConsole){
-			var consoleMethods = "assert|count|debug|dir|dirxml|group|groupEnd|info|profile|profileEnd|time|timeEnd|trace|log";
-			if(bc.stripConsole == "warn"){
-				consoleMethods += "|warn";
-			}else if(bc.stripConsole == "all"){
-				consoleMethods += "|warn|error";
-			}
-			stripConsoleRe= new RegExp("console\\.(" + consoleMethods + ")\\s*\\(", "g");
-		}
-
 		compile = function(resource, text, copyright, optimizeSwitch, callback){
 			copyright = copyright || "";
 			if(stripConsoleRe && /closure/.test(optimizeSwitch)){
