Index: web/src/main/webapp/resources/scripts/FluxInterface.js
===================================================================
--- web/src/main/webapp/resources/scripts/FluxInterface.js	(revision 3547)
+++ web/src/main/webapp/resources/scripts/FluxInterface.js	(working copy)
@@ -452,7 +452,11 @@
 
         var value = getXFormsControlValue(control);
         if (value == null || value == "") {
-            new Effect.Pulsate($(foo[i].id + "-label"));
+            var label = $(foo[i].id + "-label");
+            if (label != null)
+            {
+              new Effect.Pulsate(label);
+            }
         }
 
     }
