Index: web/src/main/java/de/betterform/agent/web/upload/MonitoredOutputStream.java
===================================================================
--- web/src/main/java/de/betterform/agent/web/upload/MonitoredOutputStream.java	(revision 3547)
+++ web/src/main/java/de/betterform/agent/web/upload/MonitoredOutputStream.java	(working copy)
@@ -118,7 +118,7 @@
 
     public void write(byte b[], int off, int len) throws IOException {
         target.write(b, off, len);
-        listener.bytesRead(len - off);
+        listener.bytesRead(len);
     }
 
     public void write(byte b[]) throws IOException {
