Index: web/src/main/java/de/betterform/agent/web/WebProcessor.java
===================================================================
--- web/src/main/java/de/betterform/agent/web/WebProcessor.java	(revision 3547)
+++ web/src/main/java/de/betterform/agent/web/WebProcessor.java	(working copy)
@@ -168,6 +168,8 @@
     protected XMLEvent exitEvent = null;
     public static final String USERAGENT = "useragent";
     public static final String REQUEST_URI = "requestURI";
+    public static final String REQUEST_URL = "requestURL";
+    public static final String QUERY_STRING = "queryString";
     public static final String CONTEXTROOT = "contextroot";
     private XFormsProcessor xformsProcessor;
     private String uploadDestination;
Index: web/src/main/java/de/betterform/agent/web/session/impl/AbstractXFormsSession.java
===================================================================
--- web/src/main/java/de/betterform/agent/web/session/impl/AbstractXFormsSession.java	(revision 3547)
+++ web/src/main/java/de/betterform/agent/web/session/impl/AbstractXFormsSession.java	(working copy)
@@ -375,6 +375,8 @@
         processor.setContextParam(XFormsProcessor.SUBMISSION_RESPONSE, servletMap);
 
         processor.setContextParam(WebProcessor.REQUEST_URI, WebUtil.getRequestURI(request));
+        processor.setContextParam(WebProcessor.REQUEST_URL, request.getRequestURL().toString());
+        processor.setContextParam(WebProcessor.QUERY_STRING, (request.getQueryString() != null ? request.getQueryString() : ""));
         processor.setContextParam(WebProcessor.CONTEXTROOT, request.getContextPath());
         if (LOGGER.isDebugEnabled()) {
             LOGGER.debug("context root of webapp: " + request.getContextPath());
