Index: web/src/main/webapp/resources/scripts/xforms-util.js
===================================================================
--- web/src/main/webapp/resources/scripts/xforms-util.js	(revision 3547)
+++ web/src/main/webapp/resources/scripts/xforms-util.js	(working copy)
@@ -153,9 +153,9 @@
             value = widget.value;
         }
     }else if (_hasClass(xformsControl,"output")){
-        if (widget.type.toLowerCase() == "a") {
+        if (widget.tagName.toLowerCase() == "a") {
             value = widget.href;
-        }else if (widget.type.toLowerCase() == "span"){
+        }else if (widget.tagName.toLowerCase() == "span"){
             value = widget.innerText;
         }
     }else if (_hasClass(xformsControl,"range")){
