Index: core/src/main/java/de/betterform/xml/xforms/xpath/saxon/function/XFormsFunctionLibrary.java
===================================================================
--- core/src/main/java/de/betterform/xml/xforms/xpath/saxon/function/XFormsFunctionLibrary.java	(revision 3547)
+++ core/src/main/java/de/betterform/xml/xforms/xpath/saxon/function/XFormsFunctionLibrary.java	(working copy)
@@ -258,6 +258,8 @@
             e = register("{" + NamespaceConstants.XFORMS_NS + "}is-card-number", IsCardNumber.class, 0, 1, 1, BuiltInAtomicType.BOOLEAN, StaticProperty.EXACTLY_ONE);
             arg(e, 0, BuiltInAtomicType.STRING, StaticProperty.EXACTLY_ONE);
 
+            e = register("{" + NamespaceConstants.XFORMS_NS + "}context", Context.class, 0, 0, 0, Type.ITEM_TYPE, StaticProperty.EXACTLY_ONE);
+
             e = register("{" + NamespaceConstants.XFORMS_NS + "}count-non-empty", CountNonEmpty.class, 0, 1, 1, BuiltInAtomicType.INTEGER, StaticProperty.EXACTLY_ONE);
             arg(e, 0, BuiltInAtomicType.ANY_ATOMIC, StaticProperty.ALLOWS_ZERO_OR_MORE);
 
