Index: web/src/main/java/de/betterform/agent/web/session/impl/XFormsSessionBase.java
===================================================================
--- web/src/main/java/de/betterform/agent/web/session/impl/XFormsSessionBase.java	(revision 3547)
+++ web/src/main/java/de/betterform/agent/web/session/impl/XFormsSessionBase.java	(working copy)
@@ -442,9 +442,17 @@
         //remove session from XFormsSessionManager
         getManager().deleteXFormsSession(this.key);
 
+        // get error page url from configuration
+        String errorpage = configuration.getProperty(WebFactory.ERROPAGE_PROPERTY);
+
+        // if absolute url, then send as is, otherwise it is relative to context root
+        if (!errorpage.startsWith("http"))
+        {
+            errorpage = request.getContextPath() + "/" + errorpage;
+        }
+
         // redirect to error page (after encoding session id if required)
-        response.sendRedirect(response.encodeRedirectURL(request.getContextPath() + "/" +
-                configuration.getProperty(WebFactory.ERROPAGE_PROPERTY)));
+        response.sendRedirect(response.encodeRedirectURL(errorpage));
     }
 
 
